packages/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp

```diff
@@ -149,12 +139,33 @@ YogaLayoutableShadowNode::YogaLayoutableShadowNode(
         static_cast<const YogaLayoutableShadowNode&>(sourceShadowNode)
             .yogaTreeHasBeenConfigured_;
   }
+                        
+  bool wasDisplayContents = hasDisplayContentsStyle();
 
   if (fragment.props) {
     updateYogaProps();
   }
 
-  if (fragment.children) {
+  bool isDisplayContents = hasDisplayContentsStyle();
+                        
+  if (!getTraits().check(ShadowNodeTraits::Trait::LeafYogaNode) && !isDisplayContents) {
+    for (auto& child : getChildren()) {
+      if (auto layoutableChild = std::dynamic_pointer_cast<const YogaLayoutableShadowNode>(child)) {
+        runForEveryConcreteSubtree(layoutableChild, [&](const YogaLayoutableShadowNode::Shared subtreeRoot) {
+          yogaLayoutableChildren_.push_back({
+            .directChild = layoutableChild,
+            .yogaChild = subtreeRoot,
+          });
+        });
+      }
+    }
+  }
+  
+  if (isDisplayContents) {
+    yogaNode_.setChildren({});
+  }
+
+  if (fragment.children || (!isDisplayContents && wasDisplayContents)) {
```

Comment:
If the node had `display: contents` but doesn't anymore, its children need to be updated since the list wasn't initialized and they were mounted higher up the tree.